spec/models/user_spec.rb

```diff
@@ -0,0 +1,12 @@
+# spec/models/user_spec.rb
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { is_expected.to validate_presence_of(:name) }
+  it { is_expected.to validate_numericality_of(:posts_counter).only_integer.is_greater_than_or_equal_to(0) }
+
+  # Additional validation tests
+  it { is_expected.to have_many(:posts).with_foreign_key('author_id') }
+  it { is_expected.to have_many(:comments) }
+  it { is_expected.to have_one(:like) }
+end
```

Comment:
- [ ] 👍 Great progress on incorporating tests for your models! However, I observed that currently, all the model tests are failing. 🚨 I recommend taking a closer look at these failures and making the necessary adjustments to ensure that all tests pass successfully.

- [ ] Additionally, I noticed that there are custom methods in your user model that lack corresponding tests. 🤔 According to the project requirements, it's crucial to test all custom methods within the models. Please prioritize writing tests for these methods to ensure comprehensive test coverage.

<img width="593" alt="Screenshot 2024-01-13 at 21 26 11" src="https://github.com/enocol/Blogger/assets/107192076/c68a1118-a734-494a-8299-056ff438c7cc">



Happy coding! ✅🚀😊